scripts/update_lti.js

```diff
@@ -0,0 +1,254 @@
+          results.notfound = results.notfound.concat(toolsToUpdate);
+          return callback();
+        }
+
+        return updateToolsForAccount(course.canvas, courseProperties.account_id, toolsToUpdate, callback);
+      });
+    });
+  });
+};
+
+/**
+ * Update SuiteC tools associated with a Canvas account.
+ *
+ * @param  {Canvas}           canvas              The Canvas object to update
+ * @param  {Number}           accountId           The id of the account to update
+ * @param  {Object[]}         toolsToUpdate       Tools to be updated 
+ * @param  {Function}         callback            Standard callback function
+ */
+var updateToolsForAccount = function(canvas, accountId, toolsToUpdate, callback) {
+  CanvasAPI.getExternalToolsForAccount(canvas, accountId, function(err, accountTools) {
+    if (err) {
+      log.error({'err': err, 'account': account.id, 'accountId': accountId}, 'Failed to get external tool configurations');
+      results.error = results.error.concat(toolsToUpdate);
+      return callback();
+    }
+
+    async.eachOfSeries(toolsToUpdate, function(tool, index, done) {
+      // If the tool doesn't show under this account, it may be configured under a higher-level account. Move on to
+      // the next tool.
+      if (!_.find(accountTools, {'id': tool.id})) {
+        return done();
+      }
+
+      // If we've found the tool configuration, attempt to update.
+      CanvasAPI.updateExternalToolForAccount(canvas, accountId, tool.id, tool.name, argv.suitec, function(err, data) {
+        if (err) {
+          log.error({'err': err, 'account': accountId, 'tool': tool}, 'Failed to update tool');
+          results.error.push(tool);
+        } else {
+          log.info({'tool': tool, 'account': accountId}, 'Updated tool at the account level');
+          results.success.push(tool);
+        }
+
+        // Whether we've succeeded or errored, we shouldn't try this tool again. Null out the array at this index.
+         toolsToUpdate[index] = null;
+        return done();
+      });
+
+    }, function() {
+      // Remove elements that were nulled out during iteration.
+      toolsToUpdate = _.compact(toolsToUpdate);
+      if (_.isEmpty(toolsToUpdate)) {
+        return callback();
+      }
+
+      // If we still have tools to update, get the parent account ID and look for configurations there.
+      CanvasAPI.getAccountProperties(canvas, accountId, function(err, accountProperties) {
+        if (err) {
+          log.error({'err': err, 'account': accountId}, 'Failed to get account properties');
+          results.error = results.error.concat(toolsToUpdate);
+          return callback();
+        }
+        if (!accountProperties.parent_account_id) {
+          // If we can't find a parent account ID, mark the remaining tools as not found.
+          results.notfound = results.notfound.concat(toolsToUpdate);
+          return callback();
+        }
```

Comment:
@johncrossman Ah, I didn't know about that endpoint. I agree we should verify, but perhaps best done in an advance step separate from this script? I think confusing Canvas terminology means that "account" in the endpoint refers to a user account (e.g. the SuiteC SPA), whereas "account" in `updateToolsForAccount` refers to a subaccount (e.g. "Official Courses").